components/cdc/src/endpoint.rs

```diff
@@ -1253,6 +1252,28 @@ impl Initializer {
             error!("cdc schedule task failed"; "error" => ?e);
         }
     }
+
+    fn deregister_downstream(&self, err: Error) {
+        let deregister = if self.build_resolver || matches!(err, Error::Request(_)) {
```

Comment:
Can you add some comments on the conditions used here?

Reply:
Do you mean comments in line 1258?

Reply:
It would be better if you could write something on how/when to use this function and how it is related to the conditions used here. I feel this could benefit those who would like to rewrite/modify this file in the future.